### PR TITLE
Enhance github-review-requests extension

### PR DESCRIPTION
**Pull request by Hephaest**
## Description

Hi guys, this PR is mainly for menu bar UI revamp that could boost engineer's working efficiency. To make this extension more flexible, I add a new preference to filter PRs by organizations/owners.

I knew this code change could be quite a bit but please consider it hasn't been updated for almost 2 years. To make reviewers more easily understand what I want to improve, I would like to give more detailed background information:

> As a software engineer, I work on multiple repos at the same time. It's difficult to catch up on what I haven't handled in time. I would like to check all my PR progress from time to time, and most importantly, I don't need to think about what's the next step since each sub-menu could indicate what it is.

My team members love it and I strongly believe this could dramatically boost engineer's working efficiency. Therefore, I made some contributions to Raycast community and hope everyone can benefit from this!

## Screencast

Menu Bar - not empty state
image
image

Menu Bar - empty state
image

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/github-review-requests/package.json [hunk 1]
```diff
@@ -32,34 +37,45 @@
...
       "title": "Personal Access Token",
       "description": "Create a token with repository access"
+    },
+    {
+      "name": "owners",
+      "type": "textfield",
+      "required": false,
+      "title": "Organizations/Owners",
+      "description": "Please specify the organizations or owners you want to focus on, separating them by commas.",
+      "placeholder": "(Optional) Include all orgs or owners."
```
Comment (kudrykv) [inline extensions/github-review-requests/package.json:47]: Maybe, somehow reword it, signifying that it's okay to leave it empty, and in this case, one would get all owners in the response.

File: extensions/github-review-requests/src/actionablePullRequests.tsx [hunk 2]
```diff
@@ -1,60 +1,132 @@
...
-      )}
+      </MenuBarExtra.Section>
+      <>
+        {Object.entries(pullsByOwner).map(([owner, pulls]) => {
+          if (pulls.length === 0) {
+            return null;
+          }
 
-      {forceRefreshSeparator && <MenuBarExtra.Separator />}
+          // Pulls
```
Comment (kudrykv) [inline extensions/github-review-requests/src/actionablePullRequests.tsx:42]: The code below is self-explanatory. I think we don't need comments here

File: extensions/github-review-requests/src/actionablePullRequests.tsx [hunk 3]
```diff
@@ -1,60 +1,132 @@
...
+          const wfmPulls = pulls.filter(
```
Comment (kudrykv) [inline extensions/github-review-requests/src/actionablePullRequests.tsx:43]: Not a fan of "wfm", "wfc", "wfr", "nrr". It takes five minutes to forget what that is. It's best to give it a proper name

File: extensions/github-review-requests/src/util.ts [hunk 4]
```diff
@@ -1,3 +1,4 @@
+import _get from "lodash/get";
```
Comment (kudrykv) [inline extensions/github-review-requests/src/util.ts:1]: Should we add only `lodash/get` to dependencies if only the `_get` is used?

**Review by kudrykv — COMMENTED** [review]
This lengthens the mouse path to get to PRs. Still, this might be more pleasant to use. It could be fun to try it out

**Comment by Hephaest** [comment]
Dear @kudrykv, I really appreciated your code review. You're right, it's essential to keep the code clean and human-readable. I made a few code changes to align our goals and resolved your comments. Could you please take some time to look through it again? 

This would be very helpful 🙏 ❤️

**Review by kudrykv — APPROVED** [review]
Looks good to me, thank you!
Now you need to find someone from the Raycast team to review, check, and merge 🙂

**Comment by Hephaest** [comment]
Hello @pernielsentikaer, would you be available to review my code when you have a moment? Thank you very much!

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Comment by akn-trackunit** [comment]
Does this update require a token with different permissions? Since the update I cannot see PR review requests.

**Comment by kudrykv** [comment]
@Hephaest could you please check out the issue with numbering?
CleanShot 2024-03-18 at 14 05 40

**Comment by Hephaest** [comment]
I'm really sorry, @kudrykv I knew the root cause of this bug and was just on the way home 😢 I will create a new PR to fix this ASAP! 🙇🏻‍♀️

**Comment by Hephaest** [comment]
Dear @kudrykv , @pernielsentikaer has helped us merge the bugfix in this [PR](https://github.com/raycast/extensions/pull/11353).

Thank you again!! @pernielsentikaer ❤️

**Comment by kudrykv** [comment]
Thank you @Hephaest!

**Pull request merged**